test/metabase/lib/native_test.cljc

```diff
@@ -314,6 +314,65 @@
     (is (not (lib/can-run (update-in (lib/native-query (metadata-provider-requiring-collection) "select * {{foo}}" nil {:collection "foobar"})
                                      [:stages 0] dissoc :collection))))))
 
+(deftest ^:parallel can-save-native-test
+  (is (lib/can-save (lib/native-query meta/metadata-provider "select 1")))
+  (is (lib/can-save (lib/native-query meta/metadata-provider "select * {{foo}}")))
+  (is (lib/can-save (lib/with-template-tags
+                      (lib/native-query meta/metadata-provider "select * {{foo}}")
+                      {"foo" {:type :text
+                              :id "1"
+                              :name "foo"
+                              :display-name "foo"}})))
```

Comment:
I think this would be easier to read with a helper function, since there's a low "signal-to-noise" ratio in these tests. It's tricky to spot the differences between them.

I'd define
```clojure
(defn- query-with-default [default]
  (lib/with-template-tags
   (lib/native-query meta/metadata-provider "select * {{foo}}")
   {"foo" {:type :text
           :id "1"
           :name "foo"
           :display-name "foo"
           :required true
           :default default}}))
```
and then the tests become
`(is (lib/can-save (query-with-default "A")))` and so on. Setting `:default nil` works for this first case, where we don't want to set the default.

(The first few with simpler queries and no template tag can stay as they are.)